dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/factory.py

```diff
@@ -512,52 +522,195 @@ def build_call_functions(self):
                         body_tmp.append(line)
 
                 list_body_to_analyse.append(body_tmp)
-                body_to_add.append(False)
-
-        # functions calling an external function ???
-        num_body = 0
-        for body in list_body_to_analyse:
-            for name in name_func_to_search:
-                if any(name in line for line in body):
-                    body_to_add[num_body] = True
-                    break
 
-            num_body += 1
+        global_pattern_index = 0
+        dic_var_name_to_temporary_name = {}
 
-        # Add the functions concerned
-        num_body = 0
+        # functions calling an external function ???
         for body in list_body_to_analyse:
-            if body_to_add[num_body]:
-                self.list_call_for_setf.append('  {\n')
+            new_body = []
+            is_discrete = False
+            found = False
+            for line in body:
+                for name in name_func_to_search:
+                    if name+"(" in line or name +" (" in line:
+                        for func in filter(lambda x: (x.get_name() == name), list_omc_functions):
+                            variables_set_by_omc_function = self.find_variables_set_by_omc_function(line, func, None)
+                            discrete_variables_set_by_omc_function = self.find_output_integer_double_vars(line, func)
```

Comment:
If I understand it right, the find_output_integer_double_vars is just filtering discrete vars from variables_set_by_omc_function. 
In order to easily understand the next test, perhaps we could have done the filtering directly there on variables_set_by_omc_function

Reply:
the method find_output_integer_double_vars is called also at another place. However, I am going to rename it (find_discrete_variables_set_by_omc_function)